be/src/util/defer_op.h

```diff
@@ -34,6 +34,24 @@ class DeferOp {
     DeferFunction _func;
 };
 
+// A Better Defer operator #5576
+// for C++17
+// Defer defer {[]{ call something }};
+//
+// for C++11
+// auto op = [] {};
+// Defer<decltype<op>> (op);
+template <class T>
+class Defer {
+public:
+    Defer(T& closure) : _closure(closure) {}
```

Comment:
may only Defer(T&& closure) : _clousure(std::forward<T>(closure)) is better

Reply:
std::forward<T>(closure) will got a compile error
such as:
```
error: cannot bind rvalue reference of type 'main()::<lambda()>&&' to lvalue of type 'main()::<lambda()>'
```

Reply:
I test the code，do not have the compile error.